docs/tutorial/go.rst

```diff
@@ -0,0 +1,361 @@
+========================================
+
+As a final step, let's update our application. For example,
+we want to add a new ``/time`` endpoint which returns the current time.
+
+Start by opening the ``main.go`` file in a text editor and update the code to
+look like the following:
+
+.. literalinclude:: code/go/main.go.time
+    :language: go
+
+Since we are creating a new version of the application, open the
+``rockcraft.yaml`` file and change the ``version`` (e.g. to ``0.2``).
+
+.. note::
+
+    ``rockcraft pack`` will create a new image with the updated code even if we
+    don't change the version. It is recommended to change the version whenever
+    we make changes to the application in the image.
+
+Pack and run the rock using similar commands as before:
+
+.. literalinclude:: code/go/task.yaml
+    :language: text
+    :start-after: [docs:docker-run-update]
+    :end-before: [docs:docker-run-update-end]
+    :dedent: 2
+
+.. note::
+
+    Note that the resulting ``.rock`` file will now be named differently, as
+    its new version will be part of the filename.
+
+Finally, use ``curl`` to send a request to the ``/time`` endpoint:
+
+.. literalinclude:: code/go/task.yaml
+    :language: text
+    :start-after: [docs:curl-time]
+    :end-before: [docs:curl-time-end]
+    :dedent: 2
+
+The updated application should respond with the current date and time (e.g.
+``2024-06-21 09:47:56``).
+
+.. note::
+
+    If we are not getting the current date and time from the ``/time``
+    endpoint, check the :ref:`troubleshooting-go` steps below.
+
+
+Cleanup
+~~~~~~~
+
+We can now stop the container and remove the corresponding image:
+
+.. literalinclude:: code/go/task.yaml
+    :language: bash
+    :start-after: [docs:stop-docker-updated]
+    :end-before: [docs:stop-docker-updated-end]
+    :dedent: 2
+
+
+Reset the environment
+======================
+
+We've reached the end of this tutorial.
+
+If we'd like to reset the working environment, we can simply run the
+following:
+
+.. literalinclude:: code/go/task.yaml
+    :language: bash
+    :start-after: [docs:cleanup]
+    :end-before: [docs:cleanup-end]
+    :dedent: 2
+
+.. collapse:: If using Multipass...
+
+    If we created an instance using Multipass, we can also clean it up.
+    Start by exiting it:
+
+    .. code-block:: bash
+
+        exit
+
+    And then we can proceed with its deletion:
+
+    .. code-block:: bash
+
+        multipass delete rock-dev
+        multipass purge
+
+----
+
+Next steps
+==========
+
+* :ref:`Rockcraft tutorials<tutorial>`.
+* :ref:`go-framework reference<go-framework-reference>`.
+* :ref:`why_use_rockcraft`.
+* :ref:`What is a Rock?<rocks_explanation>`.
+
+----
+
+.. _troubleshooting-go:
+
+Troubleshooting
+===============
+
+**Application updates not taking effect?**
+
+Upon changing the Go application and re-packing the rock, if
+the changes are not taking effect, try running ``rockcraft clean`` and pack
+the rock again with ``rockcraft pack``.
```

Comment:
Instead of linking here, I feel it would be better to simply instruct the user to run `rockcraft clean` prior to repacking the rock and remove this section. Even if `clean` isn't always required, it helps the reader build muscle memory of the commands they'll often need to reach for when iteratively building and testing.